explainer.md

```diff
@@ -0,0 +1,329 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Background: assistive technology and the accessibility tree](#background-assistive-technology-and-the-accessibility-tree)
+  - [Accessibility node properties](#accessibility-node-properties)
+- [Existing accessibility APIs](#existing-accessibility-apis)
+  - [Native platform accessibility APIs](#native-platform-accessibility-apis)
+  - [Gaps in the web platform's accessibility story](#gaps-in-the-web-platforms-accessibility-story)
+  - [Audience for the proposed API](#audience-for-the-proposed-api)
+- [The Accessibility Object Model](#the-accessibility-object-model)
+  - [Exploring the accessibility tree](#exploring-the-accessibility-tree)
+- [Use cases](#use-cases)
+  - [Use DOM node references instead of IDREFs](#use-dom-node-references-instead-of-idrefs)
+  - [Create virtual accessibility trees](#create-virtual-accessibility-trees)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Background: assistive technology and the accessibility tree
+
+Assistive technology, in this context, refers to a third party application
+which augments or replaces the existing UI for an application.
+One well-known example is a screen reader,
+which replaces the visual UI and pointer-based UI
+with an auditory output (speech and tones)
+and a keyboard and/or gesture-based input mechanism,
+completely separate from any audio output or keyboard or gesture input handled by the app.
+
+Many assistive technologies interact with a web page via accessibility APIs, such as
+[UIAutomation](https://msdn.microsoft.com/en-us/library/windows/desktop/ee684009.aspx)
+on Windows, or
+[NSAccessibility](https://developer.apple.com/library/mac/documentation/AppKit/Reference/NSAccessibility_Protocol_Reference/)
+on OS X.
+These APIs allow an application to expose a tree of objects representing the application's interface,
+typically with the root node representing the application window,
+with various levels of grouping node descendants down to individual interactive elements.
+This is referred to as the **accessibility tree**.
+
+An assistive technology user interacts with the application almost exclusively via this API,
+as the assistive technology uses it both to create the alternative interface,
+and to route user interaction events triggered by the user's commands to the assistive technology.
```

Comment:
Actual user interaction events e.g. mouse move

On Tue, Aug 30, 2016 at 3:26 PM, Rob Dodson <notifications@github.com>
wrote:

> In explainer.md
> <https://github.com/robdodson/a11y-api/pull/1#discussion_r76891565>:
>
> > +and a keyboard and/or gesture-based input mechanism,
> > +completely separate from any audio output or keyboard or gesture input handled by the app.
> > +
> > +Many assistive technologies interact with a web page via accessibility APIs, such as
> > +[UIAutomation](https://msdn.microsoft.com/en-us/library/windows/desktop/ee684009.aspx)
> > +on Windows, or
> > +[NSAccessibility](https://developer.apple.com/library/mac/documentation/AppKit/Reference/NSAccessibility_Protocol_Reference/)
> > +on OS X.
> > +These APIs allow an application to expose a tree of objects representing the application's interface,
> > +typically with the root node representing the application window,
> > +with various levels of grouping node descendants down to individual interactive elements.
> > +This is referred to as the **accessibility tree**.
> > +
> > +An assistive technology user interacts with the application almost exclusively via this API,
> > +as the assistive technology uses it both to create the alternative interface,
> > +and to route user interaction events triggered by the user's commands to the assistive technology.
>
> Sorry I wasn't clear. I was referring to "user interaction events
> triggered by users's commands" as seeming redundant. What else would
> trigger user interaction events?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/robdodson/a11y-api/pull/1/files/eefd777f0c98a6dfa3e8886bc6151a8c3b1d6cc6#r76891565>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFz6KJvcJWWhm8gxMrV7iBuetBQaMC4ks5qlK4BgaJpZM4Jtrcq>
> .
>